channels/proxies.py

```diff
@@ -59,3 +62,28 @@ def article(self):
     def article_content(self):
         """Return the article content for this post"""
         return self.article.content
+
+
+def proxy_posts(submissions):
+    """
+    Helper function to proxy submissions and posts. Posts are loaded lazily.
+
+    Args:
+        submissions (list of praw.models.Submission):
+            A list of submissions
+
+    Returns:
+        list of ProxyPost: list of proxied posts
+    """
+    # Posts are wrapped in a lazy object so we don't hit the database until something requests a property on Post
+    posts = SimpleLazyObject(lambda: {
+        post.post_id: post
+        for post in Post.objects.filter(
+            post_id__in=[submission.id for submission in submissions]
+        )
+    })
+    return [
+        PostProxy(submission, SimpleLazyObject(lambda: posts[submission.id]))
+        for submission in submissions
+        if submission.id in posts
```

Comment:
This guarantees that you would run the function passed into the `SimpleLazyObject` function, doesn't it? Seems to defeat the purpose of using `SimpleLazyObject` in the first place. I think this conditional is unnecessary anyway - the query above is already selecting for `Post`s that exist in the list of submission ids.

Also I still think it would be cleaner to (a) add a flag to the `PostProxy` constructor to fetch the `Post` for itself in a lazy object, or (b) create a `proxy_post` helper method to create a single `PostProxy`. Would  be cleaner than writing `proxy_posts([obj])[0]` every time we need a single `PostProxy`

Reply:
I opted for the latter solution, since it encapsulates construction of the proxies in that module.